Undefined symbol: ___isOSVersionAtLeast with macOS toolchains

With this code:

```swift
if #available(iOS 13.0, *) {
    print("13")
} else {
    print("before")
}
```

If you compile this with a macOS toolchain downloaded from https://www.swift.org/download, you get this error:

```
% xcrun --toolchain org.swift.57202212131a --sdk iphonesimulator swiftc /tmp/foo.swift -o /tmp/abc -target arm64-apple-ios14.0-simulator

error: link command failed with exit code 1 (use -v to see invocation)
Undefined symbols for architecture arm64:
  "___isPlatformVersionAtLeast", referenced from:
      swift::adoptTaskVoucher(swift::AsyncTask*) in libswiftCompatibility56.a(Actor.cpp.o)
      swift::restoreTaskVoucher(swift::AsyncTask*) in libswiftCompatibility56.a(Actor.cpp.o)
ld: symbol(s) not found for architecture arm64
clang-13: error: linker command failed with exit code 1 (use -v to see invocation)
```

This is because the toolchain only contains a `clang_rt` library for macOS in `./usr/lib/clang/13.0.0/lib/darwin/libclang_rt.osx.a`, vs what Xcode provides (14.2 in this example):

```
% fd clang_rt /Applications/Xcode-14.2.0-RC1.app | grep ios
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.iossim.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.ubsan_ios.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.orc_iossim.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.profile_ios.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.cc_kext_ios.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.ubsan_iossim.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.asan_ios_dynamic.dylib
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.ubsan_iossim_dynamic.dylib
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.orc_ios.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.tsan_iossim_dynamic.dylib
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.asan_iossim_dynamic.dylib
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.ios.a
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.ubsan_ios_dynamic.dylib
/Applications/Xcode-14.2.0-RC1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/14.0.0/lib/darwin/libclang_rt.profile_iossim.a
```

One workaround as discussed in this forum thread https://forums.swift.org/t/undefined-symbol-isosversionatleast/29782/16 is to copy the libraries from Xcode, into the toolchain with something like this:

```
sudo cp $(xcode-select -p)/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/*/lib/darwin/libclang_rt.*.a /Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-???.xctoolchain/usr/lib/clang/???/lib/darwin
```

Another workaround is to manually link the full path to the relevant library by adding it to your `OTHER_LDFLAGS` in Xcode if you don't want to modify the toolchain directory.

Ideally this would work out of the box for folks who are trying to test toolchains with their iOS / non-macOS projects.

rdar://104724130

--- a/utils/swift_build_support/swift_build_support/products/llvm.py
+++ b/utils/swift_build_support/swift_build_support/products/llvm.py
@@ -459,6 +459,6 @@ class LLVM(cmake_product.CMakeProduct):
 
         if self.args.llvm_install_components and system() == 'Darwin':
             clang_dest_dir = '{}{}'.format(host_install_destdir,
-                                           targets.install_prefix())
+                                           self.args.install_prefix)
             self.copy_embedded_compiler_rt_builtins_from_darwin_host_toolchain(
                 clang_dest_dir)

